### ISSUE TITLE
Method which returns nested Awaitable do not warn developers to write multiple awaits.

### ISSUE THREAD

**Issue by walterlv** [dotnet/roslyn#35232]
**Version Used**: 
Visual Studio 2019

**Steps to Reproduce**:

Assume that we have a method that returns nested awaitables:

```csharp
Task<Task> FooAsync()
{
    throw new NotImplementedException();
}
```

If developers forget to await it, C# language will warn the developers to add an `await` keyboard:

```csharp
FooAsync();
```

But if we add one `await`, the C# language will not warn the developers to add another required `await` keyword:

```csharp
await FooAsync();
```

The correct usage is:

```csharp
await await FooAsync();
```

I think that C# language should add this warning because we sometimes cannot find out this incorrect usage ourselves especially in UI applications:

```csharp
await await Dispatcher.InvokeAsync(async ()=>
{
    // Ignored codes.
});
```

**Comment by walterlv** [comment]
This issue is moved from https://github.com/dotnet/csharplang/issues/2464 to here.

**Comment by HaloFour** [comment]
I'd hope that this isn't a terribly common situation, but it seems reasonable to want a warning on an expression that would result in an awaitable, even if that expression is already `await`ing.  Although, given anything could be awaitable this seems like it could result in some spurious warnings.  For example, let's say you make `DateTime` awaitable (not suggesting you do this, but it's possible), then any time you `await` a `Task<DateTime>` you'd get an additional warning about not awaiting the `DateTime`.  An analyzer could focus on `Task<T>` and `ValueTask<T>`, though.

**Comment by walterlv** [comment]
@HaloFour I posted some code that nested awaitable commonly be created by different types of awaitables, such as WPF GUI application.

```csharp
await await Dispatcher.InvokeAsync(async ()=>
{
    // Ignored codes.
});
```

**Comment by HaloFour** [comment]
I see, so your use case isn't even the `Task<Task<T>>` situation, it's `DispatcherOperation<Task<T>>`.  It does seem a little weird that WPF doesn't supply an overload that would work better for this case, although honestly I don't see why you need `Dispatcher.InvokeAsync` at all to invoke an async method.  Why not just invoke the async method directly?

**Comment by walterlv** [comment]
@HaloFour `async` methods do not mean that all code is asynchronous. In fact, the code before the first await is run synchronously. We use `Dispatcher.InvokeAsync` to dispatch it to other timings of the UI.

**Comment by HaloFour** [comment]
@walterlv 

Of course, but it's a little weird for the synchronous component of the async method to do anything that might block, so dispatching that to a separate thread seems unnecessary.  I also don't know what WPF's Dispatcher gets you over using TPL which might get you an API a little more suited for async methods.

I'm just curious because this seems like a very specific case.  In all of the async code I've written on numerous platforms (including WPF, but it's been a _very_ long time), I don't think I've ever run into a situation where I needed to unwrap a task.
